### PR TITLE
[Cleanup] remove masternode broadcast from CActiveMasternode class

### PR DESCRIPTION
**Pull request by random-zebra**
Based on top of
- [x] #1698 

This removes unused/duplicated code in `CActiveMasternode` (ref: https://github.com/PIVX-Project/PIVX/pull/1698#pullrequestreview-439245589).
It also removes the RPC `masternodedebug`, which is a buggy simplified version of `getmasternodestatus`.

### CODE REVIEW

File: src/activemasternode.cpp [hunk 1]
```diff
@@ -86,51 +86,6 @@ void CActiveMasternode::ManageStatus()
...
         }
         pnode->Release();
-
-        // Choose coins to use
-        CPubKey pubKeyCollateralAddress;
-        CKey keyCollateralAddress;
-
-        if (GetMasterNodeVin(vin, pubKeyCollateralAddress, keyCollateralAddress)) {
-            if (GetInputAge(vin) < MASTERNODE_MIN_CONFIRMATIONS) {
-                status = ACTIVE_MASTERNODE_INPUT_TOO_NEW;
```
Comment (furszy) [inline src/activemasternode.cpp:96]: with this removal, this state `ACTIVE_MASTERNODE_INPUT_TOO_NEW ` will never be set.

If it's not used anywhere, we can remove it from the states list.

Comment (random-zebra) [reply src/activemasternode.cpp:96]: Actually, I think it would be better to restore the age check (it's useful for the operator in that case).

**Comment by random-zebra** [comment]
Rebased on master and added two commits:
- e6a1a51fb52ead330f426b553aa3c9d01a84fe33 refactors `CActiveMasternode::vin` using the `Optional<>` class, as the input is set only during `EnableHotColdMasternode()` (either upon receiving the broadcast or during update). So having the vin un-initialized is a valid state, and we should check it against `nullopt` where it's used.
- 1365ee91df71f422e4a886341fa3c0f50c3a0405 addresses @furszy's comment, removing the `ACTIVE_MASTERNODE_INPUT_TOO_NEW` state.

**Comment by furszy** [comment]
Has a small conflict.

**Comment by random-zebra** [comment]
Rebased.

File: src/activemasternode.h [hunk 2]
```diff
@@ -35,14 +35,15 @@ class CActiveMasternode
...
-    CTxIn vin;
+    Optional<CTxIn> vin;
     CService service;
 
     int status;
     std::string notCapableReason;
 
     CActiveMasternode()
     {
+        vin = nullopt;
```
Comment (furszy) [inline src/activemasternode.h:45]: would be better to move it to the direct member initializer.

File: src/activemasternode.cpp [hunk 3]
```diff
@@ -113,6 +113,11 @@ std::string CActiveMasternode::GetStatus()
 
 bool CActiveMasternode::SendMasternodePing(std::string& errorMessage)
 {
+    if (!vin.is_initialized()) {
```
Comment (furszy) [inline src/activemasternode.cpp:115]: let's better compare it against `nullopt` to know whether is initialized or not. 

The main goal of the Optional class is to move away from boost internal methods, to make smoother the transition to c++17 std::optional.

File: src/masternode-budget.cpp [hunk 4]
```diff
@@ -2208,7 +2216,7 @@ void CFinalizedBudget::SubmitVote()
         return;
     }
 
-    CFinalizedBudgetVote vote(activeMasternode.vin, GetHash());
+    CFinalizedBudgetVote vote(activeMasternode.vin.get(), GetHash());
```
Comment (furszy) [inline src/masternode-budget.cpp:2219]: Better to change `activeMasternode.vin.get()` for `*activeMasternode.vin`

File: src/activemasternode.cpp (hunk 3 above)
Comment (random-zebra) [reply src/activemasternode.cpp:115]: fair enough.

**Comment by random-zebra** [comment]
Those are not necessarily related to the current cleanup. The getBalance output, for example, is used to check whether this is a hot or cold masternode, and to remove isLocked, we should first ensure that there's no other part in the masternode flow requiring an unlocked wallet.
I think that the `CActiveMasternode::ManageStatus` method would benefit from a complete overhaul, but that's out of scope for this particular PR, which is just for the removal of the masternode broadcast code duplication.
The whole flow and relative class will be changed with deterministic masternodes anyway.

**Comment by furszy** [comment]
Ok np on doing it in another PR, impossible to disagree on the complete overhaul. Went into this direction because we removed the "automatic" hot MN startup on this PR.

**Pull request merged**